pkg/common/cns-lib/vsphere/virtualcenter.go

```diff
@@ -179,9 +183,16 @@ func (vc *VirtualCenter) newClient(ctx context.Context) (*govmomi.Client, error)
 	}
 
 	s, err := client.SessionManager.UserSession(ctx)
-	if err == nil {
-		log.Infof("New session ID for '%s' = %s", s.UserName, s.Key)
+	// If session is NotAuthenticated, SessionManager.UserSession returns
+	// nil session with nil error
+	if s == nil && err == nil {
+		return nil, ErrSessionUnauthenticated
+	}
+	if err != nil {
+		log.Errorf("failed to get UserSession. Session may be NotAuthenticated. err: %v", err)
```

Comment:
Is this error message accurate? I think if session is not authenticated, then it will be returned from line 189? Right?

Reply:
Other than that, it looks good  to me.

Reply:
done. just returned error with this message

```golang
	if err != nil {
		log.Errorf("failed to get UserSession. err: %v", err)
		return nil, err
	}
```